plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,701 @@
+#!/usr/bin/python
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://wwww.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: s3_object_info
+version_added: 5.0.0
+short_description: Gather information about objects in S3
+description:
+    - Describes objects in S3.
+author:
+  - Mandar Vijay Kulkarni (@mandar242)
+options:
+  bucket_name:
+    description:
+      - The name of the bucket that contains the object.
+    required: true
+    type: str
+  object_name:
+    description:
+      - The name of the object.
+      - If not specified, a list of all objects in the specified bucket will be returned.
+    required: false
+    type: str
+  object_details:
+    description:
+      - Retrieve requested S3 object detailed information.
+    required: false
+    type: dict
+    suboptions:
+      object_acl:
+        description:
+          - Retreive S3 object ACL.
+        required: false
+        type: bool
+        default: false
+      object_legal_hold:
+        description:
+          - Retreive S3 object legal_hold.
+        required: false
+        type: bool
+        default: false
+      object_lock_configuration:
+        description:
+          - Retreive S3 object lock_configuration.
+        required: false
+        type: bool
+        default: false
+      object_retention:
+        description:
+          - Retreive S3 object retention.
+        required: false
+        type: bool
+        default: false
+      object_tagging:
+        description:
+          - Retreive S3 object Tags.
+        required: false
+        type: bool
+        default: false
+      object_attributes:
+        description:
+          - Retreive S3 object attributes.
```

Comment:
Can you add a note here that this option requires specific botocore version?

Reply:
Fixed.